third-party/rsutils/include/rsutils/deferred.h

```diff
@@ -0,0 +1,55 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include <functional>
+
+
+namespace rsutils {
+
+
+// RAII for a function: calls it on destruction, allowing automatic calls when going out of scope very similar to a
+// smart pointer that automatically deletes when going out of scope.
+//
+class deferred
+{
+public:
+    typedef std::function< void() > fn;
+
+private:
+    fn _deferred;
+
+    deferred( const deferred & ) = delete;
+    deferred & operator=( const deferred & ) = delete;
+
+public:
+    deferred( fn && f )
```

Comment:
I think that you should not handle only rvalues (moved functions), i.e., support deferred calls to member funcitons.